File: javascript/easy/validAnagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    let map = new Map();

    for(let c of s) {
        let count = map.get(c);
        map.set(c, (count === undefined) ? 1 : count+1);
    }

    for(let c of t) {
        let count = map.get(c);
        map.set(c, (count === undefined) ? -1 : count-1);
        if(map.get(c) === 0) {
            map.delete(c);
        }
    }

    return map.size === 0;
};

File: ruby/easy/valid_anagram.rb
# @param {String} s
# @param {String} t
# @return {Boolean}
def is_anagram(s, t)
    set = Array.new
    26.times do |i|
        set[i] = 0
    end
    s.each_char do |c|
        set[(c.ord - 97)] += 1
    end
    t.each_char do |c|
        set[(c.ord - 97)] -= 1
    end
    set.each do |val|
        return false if val != 0
    end
    return true
end
